client/src/actions/authActions.js

```diff
@@ -1,117 +1,45 @@
-} else {
-  backendHost = "";
-}
-const API_SERVER = `${backendHost}`;
+import API from "../lib/api";
 
 // Login - get user token
 export function loginAction(userData) {
-  return {
-    [RSAA]: {
-      endpoint: `${API_SERVER}/api/users/login`,
-      method: "POST",
-      types: [
-        "REQUEST",
-        {
-          type: "SET_CURRENT_USER",
-          payload: async (_action, _state, res) => {
-            res = await res.json();
-            // Set token to localStorage
-            const { token } = res;
-            localStorage.setItem("jwtToken", token);
-            // Set token to Auth header
-            setAuthToken(token);
-            // Decode token to get user data
-            const decoded = jwt_decode(token);
-            console.log(decoded);
-            // Set current user
-            return decoded;
-          }
-        },
-        "FAILURE"
-      ],
-      body: JSON.stringify(userData),
-      headers: { "Content-Type": "application/json" }
-    }
+  return dispatch => {
+    API.post(`/users/login`, userData).then(response => {
+      const { token } = response;
+      localStorage.setItem("jwtToken", token);
+      setAuthToken(token);
+      const decoded = jwt_decode(token);
+      dispatch(setCurrentUser(decoded));
+    });
   };
 }
 
 // Register User
 export function registerUser(userData, history) {
-  console.log(userData);
-  return {
-    [RSAA]: {
-      endpoint: `${API_SERVER}/api/users/register`,
-      method: "POST",
-      types: [
-        "REQUEST",
-        {
-          type: "SET_CURRENT_USER",
-          payload: async (_action, _state, res) => {
-            const arg = "/login";
-            history.push(arg);
-            console.log(res);
-            res = await res.json();
-            // Set token to localStorage
-            const { token } = res;
-            localStorage.setItem("jwtToken", token);
-            // Set token to Auth header
-            setAuthToken(token);
-            // Decode token to get user data
-            const decoded = jwt_decode(token);
-
-            // Set current user
-            return decoded;
-          }
-        },
-        {
-          type: "GET_ERRORS",
-          payload: async (_action, _state, res) => {
-            return res.response.data;
-          }
-        }
-      ],
-      body: JSON.stringify(userData),
-      headers: { "Content-Type": "application/json" }
-    }
+  return dispatch => {
+    API.post(`/users/register`, userData).then(response => {
+      const arg = "/login";
+      history.push(arg);
+      const { token } = response;
+      localStorage.setItem("jwtToken", token);
+      setAuthToken(token);
+      const decoded = jwt_decode(token);
+      dispatch(setCurrentUser(decoded));
+    });
   };
 }
 
-// Set logged in user
+// Log user out
+export const logoutUser = () => dispatch => {
+  localStorage.removeItem("jwtToken");
+  setAuthToken(false);
+  dispatch({ type: "USER_LOGOUT" });
```

Comment:
You don't need to do this as part of this PR, but in general I'm a fan of having canonical types live somewhere as variables that get imported into their respective action creators and reducers.  It prevents typos like using "USER_LOGOUT" in your action creator but "USER_LOGSOUT" in your reducer, for example.